.doks/doks-cluster/databases.tf

```diff
@@ -0,0 +1,213 @@
+###########
+# cluster-dev
+###########
+
+resource "digitalocean_database_cluster" "postgres-dev" {
+  name = "codacy-doks-cluster-postgres-dev"
+  engine = "pg"
+  version = var.postgres_version
+  size = var.postgres_instance_type
+  region = digitalocean_kubernetes_cluster.codacy_k8s.region
+  node_count = 1
+}
+
+resource "digitalocean_database_firewall" "postgres-dev-fw" {
+  cluster_id = digitalocean_database_cluster.postgres-dev.id
+
+  rule {
+    type  = "k8s"
+    value = digitalocean_kubernetes_cluster.codacy_k8s.id
+  }
+}
+
+resource "digitalocean_database_user" "postgres-user-dev" {
+  cluster_id = digitalocean_database_cluster.postgres-dev.id
+  name       = "codacy"
+}
+
+######################
+# dev environment
+######################
+
+locals{
+    dev_dbs = { for v in var.db_names : v => v }
+}
+
+resource "digitalocean_database_db" "postgres_dev_db" {
+  for_each = local.dev_dbs
+  name  = each.key
+  cluster_id = digitalocean_database_cluster.postgres-dev.id
+  depends_on = [digitalocean_database_cluster.postgres-dev]
+}
+
+resource "digitalocean_database_connection_pool" "postgres-dev-pool" {
+  for_each = local.dev_dbs
+  cluster_id = digitalocean_database_cluster.postgres-dev.id
+  name       = "postgres-dev-pool-${each.key}"
+  mode       = var.connection_pool_mode
+  size       = var.connection_pool_size
+  db_name    = each.key
+  user       = var.postgres_default_admin
+   depends_on = [digitalocean_database_db.postgres_dev_db]
+}
+
+resource "digitalocean_database_connection_pool" "postgres-dev-pool-defaultdb" {
```

Comment:
Provisioned separately as defaultdb comes created out of the box. Therefore, I could not reuse the foreach that i used above to create all dbs plus defaultdb